### DISCUSSION TITLE
torch version problem

### DISCUSSION THREAD

**Discussion by kilometer80** [karpathy/build-nanogpt#63] [Q&A]
I wonder which torch version Karpathy used in this amazing tutorial? Cause my torch==2.1.2 and python==3.9.12 can't convert np.uint16 into torch.long.

**Comment by peterdonnelly1** [comment]
I'm wondering too.  I am using 2.4.1 and am having trouble getting ddp working.
It runs on the two GPUs just fine until the point where I introduce the loss/gradient sync code.
